contracts/comments.js

```diff
@@ -212,105 +240,180 @@ async function payOutPost(rewardPool, token, post, timestamp, params) {
-  // ensure it cannot take more of the current pool
   const postsToPayout = await api.db.find('posts',
     {
       rewardPoolId: rewardPool._id,
-      lastPayout: { $exists: false },
-      cashoutTime: { $gte: lastPostRewardTimestamp, $lte: timestamp },
+      cashoutTime: { $gte: lastClaimDecayTimestamp, $lte: endTimestamp },
     },
     maxPostsProcessedPerRound,
     0,
     [{ index: 'byCashoutTime', descending: false }]);
-  if (postsToPayout) {
-    newPendingClaims = api.BigNumber(newPendingClaims).plus(
-      postsToPayout.reduce((x, y) => x.plus(calculateWeightRshares(rewardPool, y.voteRshareSum)),
-        api.BigNumber(0)),
-    )
-      .toFixed(SMT_PRECISION, api.BigNumber.ROUND_DOWN);
-
-    // eslint-disable-next-line no-param-reassign
-    rewardPool.pendingClaims = newPendingClaims;
-
-    let deductFromRewardPool = api.BigNumber(0);
-    let lastPostCashout = lastPostRewardTimestamp;
+  let done = false;
+  let deductFromRewardPool = api.BigNumber(0);
+  let votesProcessed = 0;
+  if (postsToPayout && postsToPayout.length > 0) {
+    let limitReached = false;
     for (let i = 0; i < postsToPayout.length; i += 1) {
       const post = postsToPayout[i];
-      const totalPayoutValue = await payOutPost(rewardPool, token, post, timestamp, params);
-      deductFromRewardPool = deductFromRewardPool.plus(totalPayoutValue);
-      lastPostCashout = post.cashoutTime;
+      const postPayoutResponse = await payOutPost(rewardPool, token, post, params);
+      const { totalPayoutValue } = postPayoutResponse;
+      votesProcessed += postPayoutResponse.votesProcessed;
+      if (postPayoutResponse.done) {
+        deductFromRewardPool = deductFromRewardPool.plus(totalPayoutValue);
+      }
+      if (!postPayoutResponse.done || votesProcessed >= maxVotesProcessedPerRound) {
+        limitReached = true;
+        break;
+      }
+    }
+    if (!limitReached && postsToPayout.length < maxPostsProcessedPerRound) {
+      done = true;
     }
-    // eslint-disable-next-line no-param-reassign
-    rewardPool.lastPostRewardTimestamp = lastPostCashout;
     // eslint-disable-next-line no-param-reassign
     rewardPool.rewardPool = api.BigNumber(rewardPool.rewardPool)
       .minus(deductFromRewardPool)
       .toFixed(token.precision, api.BigNumber.ROUND_DOWN);
+  } else {
+    done = true;
+  }
+  if (done) {
+    // eslint-disable-next-line no-param-reassign
+    rewardPool.lastClaimDecayTimestamp = endTimestamp;
   }
 }
 
+async function postClaimsInInterval(params, rewardPool, start, end) {
+  const {
+    maxPostsProcessedPerRound,
+  } = params;
+  let postOffset = 0;
+  let newPendingClaims = api.BigNumber(0);
+  let postsToPayout = await api.db.find('posts',
+    {
+      rewardPoolId: rewardPool._id,
+      cashoutTime: { $gte: start, $lte: end },
+    },
+    maxPostsProcessedPerRound,
+    postOffset,
+    [{ index: 'byCashoutTime', descending: false }]);
```

Comment:
Is it possible for two records to have the same byCashoutTime?  I think it would be good to add _id as a fallback index to sort on, as in:  ``[{ index: 'byCashoutTime', descending: false }, { index: '_id', descending: false }]``

Reply:
yes, it's possible. will add an index